src/lib/chawathe98_matcher.rs

```diff
@@ -1534,46 +1534,31 @@ pub fn edgecover_apply_edit_script(new_matcher_pruning: &mut MappingStoreGraph<S
                             from_node_parent.children(&from_arena)
                                             .collect::<Vec<NodeId<FromNodeId>>>();
                         let size_of_children = inserting_parent_children.len();
-                        if get_position.unwrap() != 0 && size_of_children > get_position.unwrap() {
-                            if get_position.unwrap() > size_of_children {
-                                get_position = Some(size_of_children);
-                            }
-
-                            let mut insert_action = Insert::new(new_node,
-                                                                Some(from_node_parent),
-                                                                get_position.unwrap() as u16);
-                            // Parent exists
-                            insert_action.apply(&mut from_arena)
-                                         .expect("Cannot apply INSERT action to AST.");
-                            script.push(insert_action);
-
-                            indexes_t_f.insert(to_node.id(), new_node.id());
-                        } else {
-                            if get_position.unwrap() > size_of_children {
-                                get_position = Some(size_of_children);
-                            }
+                        // Note, we used to check: if get_position.unwrap() != 0
+                        // and size_of_children > get_position.unwrap() here.
+                        if get_position.unwrap() > size_of_children {
+                            get_position = Some(size_of_children);
+                        }
 
-                            let mut insert_action = Insert::new(new_node,
-                                                                Some(from_node_parent),
-                                                                get_position.unwrap() as u16);
-                            // Parent exists
-                            insert_action.apply(&mut from_arena)
-                                         .expect("Cannot apply INSERT action to AST.");
-                            script.push(insert_action);
+                        let mut insert_action = Insert::new(new_node,
+                                                            Some(from_node_parent),
+                                                            get_position.unwrap() as u16);
+                        // Parent exists
```

Comment:
This comment doesn't really make sense to me. It sounds like it's a comment about a conditional, but there is no conditional here...